sdk/iot/hub/src/az_iot_hub_client_twin.c

```diff
@@ -20,7 +20,7 @@ static const az_span az_iot_hub_twin_get_pub_topic = AZ_SPAN_LITERAL_FROM_STR("$
 static const az_span az_iot_hub_twin_patch_pub_topic
     = AZ_SPAN_LITERAL_FROM_STR("$iothub/twin/PATCH/properties/reported/");
 static const az_span az_iot_hub_twin_patch_sub_topic
-    = AZ_SPAN_LITERAL_FROM_STR("$iothub/twin/PATCH/properties/desired/");
+    = AZ_SPAN_LITERAL_FROM_STR("$iothub/twin/PATCH/properties/desired/#");
```

Comment:
This seems like extra space we don't need to have. How about concatenating the hashtag from az_iot_hub_client_twin_hashtag?

Reply:
fixed